docs/ExtendingHeartbeatProperties.md

```diff
@@ -0,0 +1,96 @@
+_This document was last updated 11/25/2016 and is applicable to SDK version 2.5.0-beta2._
+
+# Extending Heartbeat Properties in Application Insights #
+
+The .NET Application Insights SDKs provide a new feature called Health Heartbeat. This feature
+sends health related data as well as environment-specific information at pre-configured
+intervals. The feature will allow you to extend the properties that will be sent every interval,
+and will also allow you to set a flag denoting a healthy or unhealthy status for each property
+you add to the heartbeat payload.
+
+## A General Code Example ##
+
+In order to add the extended properties of your choice to the Health Heartbeat as a developer
+of an ITelemetryModule, you can follow the following pattern. Note that you must first add the
+properties you want to include in the payload, and you can update (via set) the vaules and health
+status of those properties for the duration of the application life cycle.
+
+To add the payload properties, aquire the IHeartbeatPropertyManager module using the internal
+`TelemetryModules` singleton. Do this in your `Initialize(TelemetryConfiguration telemetryConfig)` implementation method, and iterate through the available modules looking for an implementation of `IHeartbeatPropertyManager`:
```

Comment:
Is TelemetryModules singleton available only inside Initialize?

Reply:
No, I will adjust the language I use as to not imply this.